lib/dependencies/LoaderPlugin.js

```diff
@@ -2,60 +2,59 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var LoaderDependency = require("./LoaderDependency");
+"use strict";
 
-function LoaderPlugin() {}
-module.exports = LoaderPlugin;
-
-LoaderPlugin.prototype.apply = function(compiler) {
-	compiler.plugin("compilation", function(compilation, params) {
-		var normalModuleFactory = params.normalModuleFactory;
+const LoaderDependency = require("./LoaderDependency");
 
-		compilation.dependencyFactories.set(LoaderDependency, normalModuleFactory);
-	});
-	compiler.plugin("compilation", function(compilation) {
-		compilation.plugin("normal-module-loader", function(loaderContext, module) {
-			loaderContext.loadModule = function loadModule(request, callback) {
-				var dep = new LoaderDependency(request);
-				dep.loc = request;
-				compilation.addModuleDependencies(module, [
-					[dep]
-				], true, "lm", false, function(err) {
-					if(err) return callback(err);
+class LoaderPlugin {
 
-					if(!dep.module) return callback(new Error("Cannot load the module"));
-					if(dep.module.building) dep.module.building.push(next);
-					else next();
+	apply(compiler) {
+		compiler.plugin("compilation", (compilation, params) => {
+			const normalModuleFactory = params.normalModuleFactory;
 
-					function next(err) {
+			compilation.dependencyFactories.set(LoaderDependency, normalModuleFactory);
+		});
+		compiler.plugin("compilation", (compilation) => {
+			compilation.plugin("normal-module-loader", (loaderContext, module) => {
+				loaderContext.loadModule = function loadModule(request, callback) {
+					const dep = new LoaderDependency(request);
```

Comment:
IMO, its better to have it as `let` instead of `const` since there is a mutation in the next line.